remove nix code from runtimemd (#1168)
Plutonomicon/cardano-transaction-lib: doc/runtime.md

    # To use the same version of `nixpkgs` as we do
    nixpkgs.follows = "cardano-transaction-lib/nixpkgs";
  };

  outputs = { self, cardano-transaction-lib, nixpkgs, ... }:
    # some boilerplate
    let
      defaultSystems = [
        "x86_64-linux"
        "x86_64-darwin"
        "aarch64-linux"
        "aarch64-darwin"
      ];
      perSystem = nixpkgs.lib.genAttrs defaultSystems;

      # generate `pkgs` with CTL's overlays applied. This gives you access to
      # various additional packages, using the same versions of CTL
      nixpkgsFor = system: import nixpkgs {
        inherit system;
        overlays = [
          cardano-transaction-lib.overlays.runtime
          # This one is optional. If you set `ctl-server.enable = true;`
          # in the runtime config as shown below, you must enable this
          # overlay. `ctl-server` itself is **only** required when using
          # CTL's `applyArgs` effect
          cardano-transaction-lib.overlays.ctl-server
          # you probably want this one too, although it's not demonstrated
          # in this example
          cardano-transaction-lib.overlays.purescript
        ];
      };

>       # The configuration for the CTL runtime, which will be passed to the

Author: all nix code below this point is basically in `nix/runtime.nix` if I got this right. so I migrated the comments to there (and not to `template/ctl-scaffold/flake.nix` as suggested in the issue for this PR).

      # expression that builds the JSON file used by Arion. This value can be
      # shared between `buildCtlRuntime` and `launchCtlRuntime`, as shown below
      #
      # You can refer to the final configuration value by passing a function
      # that takes a single arugment. Alternatively, you can pass an attrset

>       # directly

Reviewer: The template should contain instructions on how to add services, with code examples commented-out.

Author: I added the instructions for the `extraServices` and `extraDockerCompose`. is that enough or should I also add also all the other services where we have default values?

Author: sorry, just realized that I must include all services, as otherwise the attrset isn't complete..

        # by default for backwards compatibility
        ctlServer = { enable = true; port = 8081; };
        postgres = {
          port = 5432;
          user = "ctxlib";
          password = "ctxlib";
          db = "ctxlib";
        };
        # These values will generate the `config.toml` required by ogmios-datum-cache
        datumCache = {
          port = 9999;
          # If you override some part of `postgres` above, `dbConnectionString`
          # is automatically updated
          dbConnectionString = nixpkgs.lib.concatStringsSep
            " "
            [
              "host=postgres"
              "port=${toString postgres.port}"
              "user=${postgres.user}"
              "dbname=${postgres.db}"
              "password=${postgres.password}"
            ];
          blockFetcher = {
            firstBlock = {
              slot = 54066900;
              id = "6eb2542a85f375d5fd6cbc1c768707b0e9fe8be85b7b1dd42a85017a70d2623d";
            };
            autoStart = true;
            startFromLast = false;
            filter = builtins.toJSON { const = true; };
          };
        };
      };
    in

    {

>       # `launchCtlRuntime` will generate a Nix expression from the provided

Reviewer: This comment should be added to template's flake.nix.

      # config, build it into a JSON file, and then run it with Arion
      #
      # Use `nix run .#<APP>` to run the services (e.g. `nix run .#ctl-runtime`)
      apps = perSystem (system: {
        ctl-runtime = (nixpkgsFor system).launchCtlRuntime runtimeConfig;
      });

      # `buildCtlRuntime` will generate a Nix expression that, when built with
      # `pkgs.arion.build`, outputs a JSON file compatible with Arion. This can
      # be run directly with Arion or passed to another derivation. Or you can
      # use `buildCtlRuntime` with `runArion` (from the `hercules-ci-effects`)
      # library
      #

>       # Use `nix build .#<PACKAGE` to build. To run with Arion (i.e. in your

Reviewer: This too.

> Here is our ctl-scaffold [flake](https://github.com/Plutonomicon/cardano-transaction-lib/blob/develop/templates/ctl-scaffold/flake.nix) that uses the `runtime` overlay to launch all of the required services.

Reviewer: ```suggestion
The project template [flake](https://github.com/Plutonomicon/cardano-transaction-lib/blob/develop/templates/ctl-scaffold/flake.nix) uses the `runtime` overlay to launch all of the required services.
```
